### PR TITLE
Enhancement of the format detection in camera gstreamer basic

### LINKED ISSUE #58: Use gst-device-monitor to improve format detection in camera gstreamer basic
Even after the improvements in #55, the available formats for the [`CameraGstreamer`](https://github.com/LaboratoireMecaniqueLille/crappy/blob/bc0c43cca3ec3499b2ac2d6687c01fe6dbd28070/src/crappy/camera/gstreamer_camera_basic.py) Camera are guessed when `v4l2-ctl` is not installed. This is of course not the most desirable behavior, as the guessed formats could be wrong and there could be missing formats.

Using the command `gst-device-monitor-1.0 Video/Source`, the available formats can be listed, at least on Linux. It should be tested whether this behavior is also available on Windows and macOS. It should also be checked how to set the format using only the commands of GStreamer.

### ToDo

- [x] Check the availability of `gst-device-monitor-1.0 Video/Source` on Windows and macOS
- [x] Check how to set a format using only the GStreamer commands
- [x] Implement parsing of the format string and add the format selection as an available setting

### PR DESCRIPTION
**Pull request by PIERROOOTT**
Currently, there is no format detection in the camera gstreamer basic and formats are guessed. The user can then just choose the image format among a non exhaustive list, even if some more are available. 
This is not optimal for the user.

In this PR, the available formats are detected by using the Gst device monitor (032f79f). Then, the list of the available formats is inspected in order to check if those formats are applicable (9daf836).

Closes #58.

### CODE REVIEW

**Review by WeisLeDocto — CHANGES_REQUESTED** [review]
Hi there,

I just tested your code on my machine and it crashed.
I tweaked the `__init__.py` file of the `crappy/camera` folder so that only the `gstreamer_camera_basic.py` file gets imported.
The code crashed with the follwing error:
```
Traceback (most recent call last):
  File "/home/weis/Codes/crappy-pr/src/crappy/blocks/meta_block/block.py", line 776, in run
    self.prepare()
  File "/home/weis/Codes/crappy-pr/src/crappy/blocks/camera.py", line 357, in prepare
    self._camera.open(**self._camera_kwargs)
  File "/home/weis/Codes/crappy-pr/src/crappy/camera/gstreamer_camera_basic.py", line 264, in open
    self.log(logging.DEBUG, f"The pipeline is {self._get_pipeline()}")
  File "/home/weis/Codes/crappy-pr/src/crappy/camera/gstreamer_camera_basic.py", line 389, in _get_pipeline
    img_format = img_format if img_format is not None else self.format
  File "/home/weis/Codes/crappy-pr/src/crappy/camera/meta_camera/camera.py", line 477, in __getattr__
    raise AttributeError(f'No attribute nor setting named {item}')
AttributeError: No attribute nor setting named format
```
The problem is that on line 244, the `'format'` setting never gets instantiated. It can therefore not be accessed on line 389, hence the bug ! The root of this bug is that the comparison on line 198 fails, as `'v4l2:/dev/video0'` is not equal to `'v4l2:None'`, and no camera gets selected.

So, basically, there are two things to change:

- First, make sure to handle the case when settings or attributes are not defined in `open` are called later in the code.
- Then, handle the case when no device is given by the user, i.e. `self._device = None`

Additionally, if several cameras are detected by `device_monitor.get_devices()`, the last detected one is used. I think it would be better to use the first detected one,  as it makes more sense for a default behavior.

**Review by WeisLeDocto — APPROVED** [review]
